### PR TITLE
return a keep when the ext starts a thread

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @andrewconner

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/mobile/MobileMessagingController.scala [hunk 1]
```diff
@@ -152,7 +152,7 @@ class MobileMessagingController @Inject() (
     contextBuilder += ("source", "mobile")
 
     messagingCommander.sendMessageAction(title, text, source, validUserRecipients, validEmailRecipients, validOrgRecipients, url, request.userId, contextBuilder.build).map {
-      case (message, threadInfo, messages) =>
+      case (message, threadInfo, _, messages) =>
```
Comment (camhashemi) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/mobile/MobileMessagingController.scala:155]: @DerekBurch @jaredpetker if you guys would like a keep object back from `POST /m/1/eliza/messages` (starting a discussion), let me know. I didn't want to worry about breaking deserializers for now.

**Comment by camhashemi** [comment]
merging so that current version is fixed

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala [hunk 2]
```diff
@@ -593,8 +593,10 @@ class MessagingCommander @Inject() (
             nonUserRecipients <- nonUserRecipientsFuture
             orgParticipants <- orgParticipantsFuture
             (thread, message) <- sendNewMessage(userId, userRecipients ++ orgParticipants, nonUserRecipients, url, title, text, source)(context)
+            discussionKeepFut = shoebox.getDiscussionKeepsByIds(userId, Set(thread.keepId))
             messagesWithBasicUser <- basicMessageCommander.getThreadMessagesWithBasicUser(thread)
             Seq(threadInfo) <- buildThreadInfos(userId, Seq(thread), url)
+            discussionKeepsById <- discussionKeepFut
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala:599]: maybe `discussionKeep <- discussionKeepFut.map(_.get(thread.keepId))`? Not sure if that's clearer.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/internal/ShoeboxController.scala [hunk 3]
```diff
@@ -464,7 +464,7 @@ class ShoeboxController @Inject() (
     implicit val payloadFormat = KeyFormat.key2Format[Id[User], Set[Id[Keep]]]("viewerId", "keepIds")
     val (viewerId, keepIds) = request.body.as[(Id[User], Set[Id[Keep]])]
 
-    val keepById = db.readOnlyReplica { implicit s => keepRepo.getByIds(keepIds) }
+    val keepById = db.readOnlyMaster { implicit s => keepRepo.getByIds(keepIds) }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/internal/ShoeboxController.scala:467]: Nice forethought.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala:599]: we don't want to blow up if you can't get a keep for whatever reason, the new ext will be able to handle not having it (for now, maybe at some point we want to)

Comment (camhashemi) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessagingCommander.scala:599]: Might be cleaner with `discussionKeepOpt = ...`

**Pull request merged**